retz-server/src/main/java/io/github/retz/db/Database.java

```diff
@@ -544,36 +544,44 @@ private void deleteApplication(Connection conn, String appid) throws SQLExceptio
     }
 
     // orderBy must not have any duplication
-    public List<Job> findFit(List<String> orderBy, int cpu, int memMB) throws IOException {
+    public List<Job> findAll(List<String> orderBy) throws IOException {
         List<Job> ret = new ArrayList<>();
         String orders = orderBy.stream().map(s -> s + " ASC").collect(Collectors.joining(", "));
         try (Connection conn = dataSource.getConnection(); //pool.getConnection();
              PreparedStatement p = conn.prepareStatement("SELECT * FROM jobs WHERE state='QUEUED' ORDER BY " + orders)) {
-            conn.setAutoCommit(true);
-
             try (ResultSet res = p.executeQuery()) {
-                int totalCpu = 0;
-                int totalMem = 0;
-
-                while (res.next() && totalCpu <= cpu && totalMem <= memMB) {
+                while (res.next()) {
                     String json = res.getString("json");
                     Job job = mapper.readValue(json, Job.class);
-
                     if (job == null) {
                         throw new AssertionError("Cannot be null!!");
-                    } else if (totalCpu + job.resources().getCpu() <= cpu && totalMem + job.resources().getMemMB() <= memMB) {
-                        ret.add(job);
-                        totalCpu += job.resources().getCpu();
-                        totalMem += job.resources().getMemMB();
-                    } else {
-                        break;
                     }
+                    ret.add(job);
                 }
             }
             return ret;
         } catch (SQLException | IOException e) {
-            throw new IOException(MessageFormat.format("Database.findFit({0}, {1}) failed", cpu, memMB), e);
+            throw new IOException("Database.findAll() failed", e);
+        }
+    }
+
+    public List<Job> findFit(List<String> orderBy, int cpu, int memMB) throws IOException {
+        List<Job> ret = new ArrayList<>();
+        int totalCpu = 0;
+        int totalMem = 0;
+
+        for (Job job: findAll(orderBy)) {
```

Comment:
With this style, regardless of the strategy 'all' or 'fit' the scheduler must fetch and iterate all over the jobs in the database, because in `findAll` it fetches all jobs into the database. Although whether current code iterates over all jobs in the database or not depends on implementation of JDBC drivers, I don't want iteration over all jobs explicitly happen.

I wish if Java had supported `yield` in Python, it have been much simpler but I don't have clean answer for this. Maybe 'all' strategy sets cpu and totalMem as `Long.MAX_VALUE` or add another if condition in while clause.